src/Utilities/matrix_of_constraints.jl

```diff
@@ -0,0 +1,357 @@
+# Constants: stored in a `Vector` or `Box` or any other type implementing:
+# `empty!`, `resize!` and `load_constants`.
+
+"""
+    load_constants(constants, offset, func_or_set)
+
+This function loads the constants of `func_or_set` in `constants` at an offset
+of `offset`. The storage should be preallocated with `resize!` before calling
+this function.
```

Comment:
What is `offset`? "Where `offset` is the number of constants that have previously been loaded"?